src/core/services/compiler/compiler.spec.js

```diff
@@ -163,21 +163,27 @@ describe('$mdCompiler service', function() {
         var called = false;
         var data = compile({
           template: 'hello',
-          controller: function($scope) {
-            expect(this.name).toBe('Bob');
-            expect($scope.$apply).toBeTruthy(); // test DI working properly
-            called = true;
-          },
+          controller: TestController,
           controllerAs: 'ctrl',
           bindToController: true,
           locals: { name: 'Bob' }
         });
+
         var scope = $rootScope.$new();
         data.link(scope);
         expect(scope.ctrl.name).toBe('Bob');
         expect(called).toBe(true);
-      }));
 
+
+        function TestController($scope) {
+          called = true;
+          expect($scope.$apply).toBeTruthy(); // Ensure that DI works properly
+        }
+
+        TestController.prototype.$onInit = function() {
+          expect(this.name).toBe('Bob');
```

Comment:
That's why I like QUnit's `assert.expect(number)` so much but Jasmine has no equivalent.

A proper test here should also probably spy on `$onInit` (with `calledThrough()`) to ensure it's not skipped.